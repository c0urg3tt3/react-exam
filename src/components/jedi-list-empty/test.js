import React from 'react'

import { shallow } from 'enzyme'

import JediListEmpty, { emptyQuoteProps } from './component'

describe('[component] JediListEmpty', () => {
  it('should render component', () => {
    const wrapper = shallow(
      <JediListEmpty>
      { ({message, quote, autor}) => {
        return (
          <div>
            <p className="message">{message}</p>
            <p className="quote">{quote}</p>
            <p className="autor">{autor}</p>
          </div>
        )
      }}
      </JediListEmpty>
    )

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('div')).toEqual(true)
    expect(wrapper.hasClass('jedi-list-empty')).toEqual(true)

    expect(wrapper.find('p')).toHaveLength(3)

    expect(wrapper.find('.message').text()).toEqual(emptyQuoteProps.message)
    expect(wrapper.find('.quote').text()).toEqual(emptyQuoteProps.quote)
    expect(wrapper.find('.autor').first().text()).toEqual(emptyQuoteProps.autor)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
