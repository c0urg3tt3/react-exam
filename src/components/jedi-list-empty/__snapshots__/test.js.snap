// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] JediListEmpty should render component 1`] = `"<div class=\\"jedi-list-empty\\"><div><p class=\\"message\\">Looks like someone executed Order 66 !</p><p class=\\"quote\\">Master sky walker there is too many of them what are we going to do?</p><p class=\\"autor\\">Padawan</p></div></div>"`;
